src/V3Options.cpp

```diff
@@ -917,639 +1108,544 @@ void V3Options::parseOptsList(FileLine* fl, const string& optdir, int argc, char
+        if (m_outputSplitCFuncs < 0) {
+            fl->v3error("--output-split-cfuncs must be >= 0: " << valp);
+        }
+    });
+    DECL_OPTION("-output-split-ctrace", CbVal, [this, fl](const char* valp) {
+        m_outputSplitCTrace = std::atoi(valp);
+        if (m_outputSplitCTrace < 0) {
+            fl->v3error("--output-split-ctrace must be >= 0: " << valp);
+        }
+    });
+    DECL_OPTION("-protect-lib", CbVal, [this](const char* valp) {
+        m_protectLib = valp;
+        m_protectIds = true;
+    });
+    DECL_OPTION("-trace-fst", CbCall, [this]() {
+        m_trace = true;
+        m_traceFormat = TraceFormat::FST;
+        addLdLibs("-lz");
+    });
+    DECL_OPTION("-trace-fst-thread", CbCall, [this, fl]() {
+        m_trace = true;
+        m_traceFormat = TraceFormat::FST;
+        addLdLibs("-lz");
+        fl->v3warn(DEPRECATED, "Option --trace-fst-thread is deprecated. "
+                               "Use --trace-fst with --trace-threads > 0.");
+        if (m_traceThreads == 0) m_traceThreads = 1;
+    });
+    DECL_OPTION("-trace-threads", CbVal, [this, fl](const char* valp) {
+        m_trace = true;
+        m_traceThreads = std::atoi(valp);
+        if (m_traceThreads < 0) fl->v3fatal("--trace-threads must be >= 0: " << valp);
+    });
+    DECL_OPTION("-trace-depth", Set, &m_traceDepth);
+    DECL_OPTION("-trace-max-array", Set, &m_traceMaxArray);
+    DECL_OPTION("-trace-max-width", Set, &m_traceMaxWidth);
+    DECL_OPTION("-U", CbPartialMatch, &V3PreShell::undef);
+    DECL_OPTION("-unroll-count", Set, &m_unrollCount);  // Undocumented optimization tweak
+    DECL_OPTION("-unroll-stmts", Set, &m_unrollStmts);  // Undocumented optimization tweak
+    DECL_OPTION("-v", CbVal, [this, &optdir](const char* valp) {
+        V3Options::addLibraryFile(parseFileArg(optdir, valp));
+    });
+    DECL_OPTION("-clk", CbVal, {this, &V3Options::addClocker});
+    DECL_OPTION("-no-clk", CbVal, {this, &V3Options::addNoClocker});
+    DECL_OPTION("-V", CbCall, [this]() {
+        showVersion(true);
+        exit(0);
+    });
+    DECL_OPTION("-version", CbCall, [this]() {
+        showVersion(false);
+        exit(0);
+    });
+    DECL_OPTION("-Wall", CbCall, []() {
+        FileLine::globalWarnLintOff(false);
+        FileLine::globalWarnStyleOff(false);
+    });
+    DECL_OPTION("-Werror-", CbPartialMatch, [this, fl](const char* optp) {
+        V3ErrorCode code(optp);
+        if (code == V3ErrorCode::EC_ERROR) {
+            if (!isFuture(optp)) fl->v3fatal("Unknown warning specified: -Werror-" << optp);
+        } else {
+            V3Error::pretendError(code, true);
+        }
+    });
+    DECL_OPTION("-Wfuture-", CbPartialMatch, [this](const char* optp) {
+        // Note it may not be a future option, but one that is currently implemented.
+        addFuture(optp);
+    });
+    DECL_OPTION("-Wno-", CbPartialMatch, [this, fl](const char* optp) {
+        if (!strcmp(optp, "context")) {
```

Comment:
Can we DECL_OPTION("-Wno-context") and still have a "Wno-*" catch-all?  Would look cleaner and help suggestions.

Another alternative perhaps best for suggestions is to get list of all warning codes from V3Error and loop over them and DECL_OPTION them all.

Reply:
2e1da1d8cdd8ba036d1f31af3c6d3d0237762264 covers `-Wno` and `-Wwarn`.

ef246611bf7dda0ca011cd4e6d7488c34b5d2ce5 improves `-language` suggestion.
96d2682940913a0787097bed6a8f6cb6040ce999 suggests `-no-opt`.